### PR TITLE
CMake autoformatting

### PR DESCRIPTION
**Pull request by wkaluza**
This would enforce a CMake code style in CI using the Python module cmake_format.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,13 +1,14 @@
-################################################################################
+#
```
Comment (ejfitzgerald) [inline CMakeLists.txt:1]: Seems a shame to loose these headers but whatever

File: CMakeLists.txt [hunk 2]
```diff
@@ -21,36 +22,42 @@ option(FETCH_ENABLE_COVERAGE        "Enable coverage generation"        OFF)
...
 
 # custom string options
-set(FETCH_DEBUG_SANITIZER "" CACHE STRING "The clang based sanitizer to be enabled")
-set(FETCH_COMPILE_LOGGING_LEVEL "info" CACHE STRING "The precompiled logging level for the project")
-
-# target architecture (to be replaced with automatic detection)
-## sysctl -a | grep machdep.cpu.features
+set(FETCH_DEBUG_SANITIZER
+    ""
+    CACHE STRING "The clang based sanitizer to be enabled")
```
Comment (ejfitzgerald) [inline CMakeLists.txt]: This is a good example of where after this change it looks worse. Can we not configure the cmake formatter in someway to adjust the line length. I don't really think we need to limit ourselves to something that is 80 characters in the build system.

File: CMakeLists.txt [hunk 3]
```diff
@@ -21,36 +22,42 @@ option(FETCH_ENABLE_COVERAGE        "Enable coverage generation"        OFF)
...
-set (FETCH_ROOT_CMAKE_DIR ${FETCH_ROOT_DIR}/cmake)
-set (FETCH_ROOT_VENDOR_DIR ${FETCH_ROOT_DIR}/vendor)
+set(FETCH_ROOT_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR})
+set(FETCH_ROOT_DIR ${CMAKE_CURRENT_SOURCE_DIR})
+set(FETCH_ROOT_CMAKE_DIR ${FETCH_ROOT_DIR}/cmake)
+set(FETCH_ROOT_VENDOR_DIR ${FETCH_ROOT_DIR}/vendor)
 
 # Default to a release build
-if (NOT CMAKE_BUILD_TYPE)
+if(NOT CMAKE_BUILD_TYPE)
```
Comment (ejfitzgerald) [inline CMakeLists.txt]: I sort of preferred the space separator here, not super strong though. If it is something we can configure that would be good.

File: CMakeLists.txt (hunk 2 above)
Comment (wkaluza) [reply CMakeLists.txt]: I increased the line width to 100 chars which took care of the worst offenders. I would ask that we don't go any higher, as that can be hard to read on a portrait monitor with a big font (like the one I need to use).

File: CMakeLists.txt (hunk 3 above)
Comment (wkaluza) [reply CMakeLists.txt]: Yup, that's easily done. We get treasures like `endif ()` as part of the package, though.

File: libs/core/tests/CMakeLists.txt [hunk 4]
```diff
@@ -16,5 +16,11 @@ add_fetch_test(byte_array_gtest fetch-core byte_array)
 add_fetch_test(random_generator_gtest fetch-core random)
 add_fetch_test(json_gtest fetch-core json/gtest/)
 add_fetch_test(core_encode_decode_gtest fetch-core tokenizer/gtest/)
-add_fetch_test(containers-tests fetch-core containers/ SLOW)
-add_fetch_test(sync_gtest fetch-core sync/ SLOW)
+add_fetch_test(containers-tests
```
Comment (ejfitzgerald) [inline libs/core/tests/CMakeLists.txt:19]: Lets make an issue to create a `add_fetch_slow_test` wrapper to avoid these in the future

**Review by ejfitzgerald — APPROVED** [review]

**Pull request merged**